pythia/datasets/processors.py

```diff
@@ -895,3 +899,288 @@ def __call__(self, item):
         ]
         caption = " ".join(tokens)
         return {"tokens": tokens, "caption": caption}
+
+
+@registry.register_processor("phoc")
+class PhocProcessor(VocabProcessor):
+    """
+    Compute PHOC features from text tokens
+    """
+    def __init__(self, config, *args, **kwargs):
+        self._init_extras(config)
+        self.config = config
+
+    def _map_strings_to_indices(self, tokens):
+        length = min(len(tokens), self.max_length)
+        tokens = tokens[:length]
+
+        phoc_dim = 604
+        output = torch.full(
+            (self.max_length, phoc_dim),
+            fill_value=self.PAD_INDEX,
+            dtype=torch.float,
+        )
+
+        for idx, token in enumerate(tokens):
+            output[idx] = torch.from_numpy(build_phoc(token))
+
+        return output
+
+
+@registry.register_processor("copy")
+class CopyProcessor(BaseProcessor):
+    """
+    Copy boxes from numpy array
+    """
+    def __init__(self, config, *args, **kwargs):
+        self.max_length = config.max_length
+
+    def __call__(self, item):
+        blob = item["blob"]
+        final_blob = np.zeros((self.max_length,) + blob.shape[1:], blob.dtype)
+        final_blob[:len(blob)] = blob[:len(final_blob)]
+
+        return {"blob": torch.from_numpy(final_blob)}
+
+
+@registry.register_processor("bert_tokenizer")
+class BertTokenizerProcessor(BaseProcessor):
```

Comment:
@apsdehal We should check if this is same or different from the BertTokenizer we have internally when merging.

Reply:
Ours is inside `processors/bert`.